File: practicas/prt4/empleado.py
from abc import ABC, abstractmethod

class Empleado(ABC):
    def __init__(self, nombre):
        self._nombre = nombre

    @abstractmethod
    def calcular_salario_mensual(self):
        pass

    def __str__(self):
        return f"Nombre: {self._nombre}"

    @property
    def nombre(self):
        return self._nombre

File: practicas/prt4/empleado.java
package practicas.prt4;

public abstract class empleado {
    protected String nombre;

    public empleado(String nombre) {
        this.nombre = nombre;
    }

    public abstract double CalcularSalarioMensual();

    @Override
    public String toString() {
        return "Nombre: " + nombre;
    }

    public String getNombre() {
        return nombre;
    }
}
